vmm/src/lib.rs

```diff
@@ -712,6 +696,47 @@ struct KernelConfig {
     cmdline_addr: GuestAddress,
 }
 
+/// A lightweight class that abstracts a stream of epoll events.
+/// The stream of events is equal to events[index..num_events),
+/// follwed by the events still left to be read from `epoll_raw_fd`.
+struct EpollStream {
+    epoll_raw_fd: RawFd,
+    events: Vec<epoll::Event>,
+    num_events: usize,
+    index: usize,
+}
+
+impl EpollStream {
+    fn new(epoll_raw_fd: RawFd) -> EpollStream {
```

Comment:
Functions that take `RawFd` as parameters should be declared unsafe.

Reply:
Okay, I've refactored this so that the unsafe-ness dissapears.